rpc/internal/system_metrics_service_test.go

```diff
@@ -0,0 +1,334 @@
+package internal
+
+import (
+	"context"
+	fmt "fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/pail"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	grpc "google.golang.org/grpc"
+)
+
+func TestAddSystemMetrics(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	env, err := createSystemMetricsEnv()
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, teardownSystemMetricsEnv(ctx, env))
+	}()
+	tempDir, err := ioutil.TempDir(".", "system-metrics-test")
+	defer func() {
+		assert.NoError(t, os.RemoveAll(tempDir))
+	}()
+	require.NoError(t, err)
+
+	conf, err := model.LoadCedarConfig(filepath.Join("testdata", "cedarconf.yaml"))
```

Comment:
If we're going to use `testdata/cedarconf.yaml` in multiple GRPC tests, can we just make it a constant and re-use that?